### PR TITLE
added script for Kafka installation and configuration on an Ubuntu ma…

### PR DESCRIPTION
**Pull request by yliu0715**
Script for Kafka installation and configuration meant for Ubuntu.

### CODE REVIEW

File: scripts/lib/install_kafka.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
```
Comment (chc5) [inline scripts/lib/install_kafka.sh]: Library scripts should not have #!/bin/bash at the top https://google.github.io/styleguide/shell.xml?showone=File_Extensions#File_Extensions

File: scripts/lib/install_kafka.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+# Complete installation and configuration of Kafka
+install() {
+  command -v java >/dev/null 2>&1 || { install_openJDK; }
+  download_and_extract_kafka_binaries
+  enable_topic_deletion
+  create_unit_file_for_zookeeper
+  create_unit_file_for_kafka
+}
+
+install;
```
Comment (chc5) [inline scripts/lib/install_kafka.sh]: Call this in install shell script

File: scripts/lib/install_kafka.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+[Install]
+WantedBy=multi-user.target"
+ 
+create_unit_file_for_zookeeper() {
+  echo ${UNIT_FILE_CONTENT_FOR_ZOOKEEPER} >> /etc/systemd/system/zookeeper.service
+}
+
+# Unit file for Kafka
+UNIT_FILE_CONTENT_FOR_KAFKA="[Unit]
```
Comment (chc5) [inline scripts/lib/install_kafka.sh]: https://google.github.io/styleguide/shell.xml?showone=Function_Location#Function_Location

File: scripts/lib/install_kafka.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  echo "\ndelete.topic.enable = true" >> ~/kafka/config/server.properties
+}
+
+#
+# Create systemd unit files for Kafka services such as starting, stopping, and restarting Kafka
+# consistently with other services
+#
+
+# Unit file for zookeeper
+UNIT_FILE_CONTENT_FOR_ZOOKEEPER="[Unit]
```
Comment (chc5) [inline scripts/lib/install_kafka.sh]: https://google.github.io/styleguide/shell.xml?showone=Function_Location#Function_Location

File: scripts/lib/install_kafka.sh [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+Restart=on-abnormal
+
+[Install]
+WantedBy=multi-user.target"
+create_unit_file_for_kafka() {
+  echo ${UNIT_FILE_CONTENT_FOR_KAFKA} >> /etc/systemd/system/kafka.service
+}
+
+# Complete installation and configuration of Kafka
+install() {
```
Comment (chc5) [inline scripts/lib/install_kafka.sh]: Rename this because there's already an install command

File: scripts/lib/install_kafka.sh (hunk 1 above)
Comment (yliu0715) [reply scripts/lib/install_kafka.sh]: Fixed

File: scripts/lib/install_kafka.sh (hunk 2 above)
Comment (yliu0715) [reply scripts/lib/install_kafka.sh]: Fixed

File: scripts/lib/install_kafka.sh (hunk 3 above)
Comment (yliu0715) [reply scripts/lib/install_kafka.sh]: Fixed

File: scripts/lib/install_kafka.sh (hunk 4 above)
Comment (yliu0715) [reply scripts/lib/install_kafka.sh]: Fixed

File: scripts/lib/install_kafka.sh (hunk 5 above)
Comment (yliu0715) [reply scripts/lib/install_kafka.sh]: Fixed

**Pull request merged**